test/extended/deployments/deployments.go

```diff
@@ -1651,11 +1658,35 @@ var _ = g.Describe("[Feature:DeploymentConfig] deploymentconfigs", func() {
 			dc, err = oc.AppsClient().AppsV1().DeploymentConfigs(oc.Namespace()).Patch(dcName, types.StrategicMergePatchType, []byte(fmt.Sprintf(`{"spec": {"replicas": %d}}`, newScale)))
 			o.Expect(err).NotTo(o.HaveOccurred())
 
-			w, err = oc.AppsClient().AppsV1().DeploymentConfigs(namespace).Watch(metav1.SingleObject(dc.ObjectMeta))
-			o.Expect(err).NotTo(o.HaveOccurred())
 			ctx3, cancel3 := context.WithTimeout(context.TODO(), deploymentChangeTimeout)
 			defer cancel3()
-			event, err = watchtools.UntilWithoutRetry(ctx3, w, func(e watch.Event) (bool, error) {
+
+			preconditionFunc := func(store cache.Store) (bool, error) {
+				_, exists, err := store.Get(&metav1.ObjectMeta{Namespace: dc.Namespace, Name: dc.Name})
+				if err != nil {
+					return true, err
+				}
+				if !exists {
+					// We need to make sure we see the object in the cache before we start waiting for events
+					// or we would be waiting for the timeout if such object didn't exist.
+					return true, kerrors.NewNotFound(schema.GroupResource{
+						Group:    "apps.openshift.io/v1",
+						Resource: "DeploymentConfig",
+					}, dc.Name)
+				}
+
+				return false, nil
+			}
+			lw := &cache.ListWatch{
+				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+					return oc.AppsClient().AppsV1().DeploymentConfigs(oc.Namespace()).List(metav1.ListOptions{})
```

Comment:
why are you dropping options?

Reply:
do we need options here? does it make a difference?